device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -34,6 +33,8 @@ public class IotHubTransport implements IotHubListener
 {
     private static final int DEFAULT_MAX_MESSAGES_TO_SEND_PER_THREAD = 10;
 
+    private static final int DEFAULT_CORRELATION_ID_LIVE_TIME = 60000;
```

Comment:
It's hard to know when exactly the callback of a particular correlating message will be completed OR if it will certainly happen (in case that the service fails to send the final message due to service bug or due to us losing connection), so setting this period as the default correlation ID live time. Check if a correlation ID has existed over this period every time while sending messages, if so, remove the ID from maps to avoid of out-of-memory.